src/studio/src/designer/backend/Services/Implementation/GiteaAPIWrapper.cs

```diff
@@ -314,6 +316,28 @@ public async Task<Branch> GetBranch(string org, string repository, string branch
             return null;
         }
 
+        /// <inheritdoc />
+        public async Task<Branch> CreateBranch(string org, string repository, string branchName)
+        {
+            string content = $"{{\"new_branch_name\":\"{branchName}\"}}";
+
+            HttpRequestMessage m = new HttpRequestMessage(HttpMethod.Post, $"repos/{org}/{repository}/branches");
+            m.Content = new StringContent(content, Encoding.UTF8, "application/json");
+
+            HttpResponseMessage response = await _httpClient.SendAsync(m);
+
+            if (response.StatusCode == HttpStatusCode.Created)
+            {
+                return await response.Content.ReadAsAsync<Branch>();
+            }
+            else
+            {
+                _logger.LogError($"//GiteaAPIWrapper // CreateBranch // Error ({response.StatusCode}) occured when creating branch {branchName} for repo {org}/{repository}");
```

Comment:
I've noticed that we use this syntax (//GiteaAPIWrapper // CreateBranch //) in some places and others not, is it a convention we're supposed to use? To me it seems a bit redundant as the logger will add both class and method by default and if we refactor we would need to remember to change these strings to correspond with the class/method.

Reply:
We've mostly used it to faster identify where the log message is from w.r.t debugging to avoid having to expand entries in AI to find out where the error occured